posthog/cdp/templates/rudderstack/template_rudderstack.py

```diff
@@ -0,0 +1,118 @@
+        'browser_version': event.properties.$browser_version,
+        'page': {
+            'host': event.properties.$host,
+            'url': event.properties.$current_url,
+            'path': event.properties.$pathname,
+            'referrer': event.properties.$referrer,
+            'initial_referrer': event.properties.$initial_referrer,
+            'referring_domain': event.properties.$referring_domain,
+            'initial_referring_domain': event.properties.$initial_referring_domain,
+        },
+        'screen': {
+            'height': event.properties.$screen_height,
+            'width': event.properties.$screen_width,
+        },
+        'library': {
+            'name': event.properties.$lib,
+            'version': event.properties.$lib_version,
+        },
+        'ip': event.$ip,
+        'active_feature_flags': event.properties.$active_feature_flags,
+        'token': event.properties.token
+    },
+    'channel': 's2s',
+    'messageId': event.uuid,
+    'originalTimestamp': event.timestamp,
+    'userId': event.properties.$user_id ?? event.distinct_id,
+    'anonymousId': event.properties.$anon_distinct_id ?? event.properties.$device_id ?? event.properties.distinct_id,
+    'type': 'track',
+    'properties': {},
+}
+
+if (event.name in ('$identify', '$set')) {
+    rudderPayload.type := 'identify'
+    rudderPayload.context.trait := event.properties.$set
+    rudderPayload.traits := event.properties.$set
+} else if (event.name == '$create_alias') {
+    rudderPayload.type := 'alias'
+    rudderPayload.userId := event.properties.alias
+    rudderPayload.previousId := event.distinct_id
+} else if (event.name == '$pageview') {
+    rudderPayload.type := 'page'
+    rudderPayload.name := event.properties.name
+    rudderPayload.properties.host := event.properties.$host
+    rudderPayload.properties.url := event.properties.$current_url
+    rudderPayload.properties.path := event.properties.$pathname
+    rudderPayload.properties.referrer := event.properties.$referrer
+    rudderPayload.properties.initial_referrer := event.properties.$initial_referrer
+    rudderPayload.properties.referring_domain := event.properties.$referring_domain
+    rudderPayload.properties.initial_referring_domain := event.properties.$initial_referring_domain
+} else if (event.name == '$autocapture') {
+    rudderPayload.type := 'track'
+    rudderPayload.event := event.properties.$event_type
+} else {
+    rudderPayload.type := 'track'
+    rudderPayload.event := event.name
+}
+
+for (let key, value in event.properties) {
+    if (value != null and not key like '$%') {
+        rudderPayload.properties[key] := value
+    }
+}
+
+let payload := {
+    'batch': [rudderPayload],
+    'sentAt': now()
+}
+fetch(f'{host}/v1/batch', {
+    'method': 'POST',
+    'headers': {
+        'Content-Type': 'application/json',
+        'Authorization': f'Basic {base64Encode(f'{inputs.token}:')}',
+    },
+    'body': payload
+})
```

Comment:
Something to keep in mind is that HogVM stores all the values you declare on a stack. So `host`, `token`, `rudderPayload`, `payload`, and the arguments to `fetch` will all be stored separately on the stack. This will work fine, but when we serialize the stack to send it all to the "fetch service", we're effectively duplicating the same data multiple times in memory. You can see the bytes used from the function's log tab.

A somewhat cleaner approach is to put all the payload generation into a function, and call that directly within the arguments to `fetch`. Whenever a function returns, all the temporary values it used get cleared from memory. So you'd write something like `fetch(..., { body: getPayload() })`. 

See the AWS kinesis function for an example where I have done just that.